client/extensions/woocommerce/app/settings/taxes/index.js

```diff
@@ -0,0 +1,203 @@
+		const option = event.target.name;
+		const value = event.target.checked;
+		this.setState( { [ option ]: value, userBeganEditing: true } );
+	}
+
+	pageHasChanges = () => {
+		return this.state.userBeganEditing;
+	}
+
+	onSave = ( event ) => {
+		const { site, translate } = this.props;
+
+		event.preventDefault();
+		this.setState( { isSaving: true } );
+
+		const onSuccess = () => {
+			this.setState( { isSaving: false, userBeganEditing: false } );
+			return successNotice( translate( 'Settings updated successfully.' ) );
+		};
+
+		const onFailure = () => {
+			this.setState( { isSaving: false } );
+			return errorNotice( translate( 'There was a problem saving your changes. Please try again.' ) );
+		};
+
+		// TODO - chain these
+
+		this.props.updateTaxesEnabledSetting(
+			site.ID,
+			this.state.taxesEnabled,
+			null,
+			null,
+		);
+
+		this.props.updateTaxSettings(
+			site.ID,
+			this.state.pricesIncludeTax,
+			! this.state.shippingIsTaxable, // note the inversion
+			onSuccess,
+			onFailure
+		);
+	};
+
+	render = () => {
+		const { className, loading, site, translate } = this.props;
+
+		if ( loading ) {
+			// TODO placeholder
+			return null;
+		}
+
+		const breadcrumbs = [
+			( <a href={ getLink( '/store/:site/', site ) }>{ translate( 'Settings' ) }</a> ),
+			( <span>{ translate( 'Taxes' ) }</span> ),
+		];
+
+		const saveButtonDisabled = this.state.isSaving || ! this.pageHasChanges();
+
+		return (
+			<Main className={ classNames( 'settings-taxes', className ) }>
+				<ActionHeader breadcrumbs={ breadcrumbs }>
+					<Button disabled={ saveButtonDisabled } onClick={ this.onSave } primary >
+						{ translate( 'Save' ) }
+					</Button>
+				</ActionHeader>
+				<TaxesNexus
+					site={ site }
+				/>
+				<TaxesRates
+					taxesEnabled={ this.state.taxesEnabled }
+					onEnabledChange={ this.onEnabledChange }
+					site={ site }
+				/>
+				<TaxesOptions
+					onCheckboxChange={ this.onCheckboxChange }
+					pricesIncludeTaxes={ this.state.pricesIncludeTaxes }
+					shippingIsTaxable={ this.state.shippingIsTaxable }
+				/>
+			</Main>
+		);
+	}
+
+}
+
+function mapStateToProps( state ) {
+	let loading = true;
+	let pricesIncludeTaxes = false;
+	let shippingIsTaxable = false;
+	let taxesEnabled = false;
+
+	const site = getSelectedSiteWithFallback( state );
+	if ( site ) {
+		loading = areTaxSettingsLoading( state, site.ID ) || areSettingsGeneralLoading( state, site.ID );
+		if ( ! loading ) {
+			pricesIncludeTaxes = getPricesIncludeTax( state, site.ID );
+			shippingIsTaxable = ! getShippingIsTaxFree( state, site.ID ); // note the inversion
+			taxesEnabled = areTaxCalculationsEnabled( state, site.ID );
+		}
+	}
+
+	return {
+		loading,
+		pricesIncludeTaxes,
+		shippingIsTaxable,
+		site,
+		taxesEnabled,
+	};
+}
+
+function mapDispatchToProps( dispatch ) {
+	return bindActionCreators(
+		{
+			fetchSettingsGeneral,
+			fetchTaxSettings,
+			updateTaxesEnabledSetting,
+			updateTaxSettings
```

Comment:
Fixed in https://github.com/Automattic/wp-calypso/pull/15541/commits/17f50964f1f6156641032d53a6b96785ad2895a0